diff --git a/dist/module.mjs b/dist/module.mjs
index 8f47951e0d9ccd8691b9d6256db5d8342945c30d..be11337fe3b67ce8e9b8c7f17490fe5653e4d80a 100644
--- a/dist/module.mjs
+++ b/dist/module.mjs
@@ -224,6 +224,9 @@ const module = defineNuxtModule({
       );
     }
     const styleContent = () => {
+      if (options?.unstyled) {
+        return `const styles = [], stylesToTop = [], themes = []; export { styles, stylesToTop, themes }`;
+      }
       if (!loadStyles)
         return `export const styles = [], stylesToTop = [], themes = [];`;
       const uniqueRegisteredStyles = Array.from(new Map(registeredStyles?.map((m) => [m.name, m])).values());
