diff --git a/tooltip/index.mjs b/tooltip/index.mjs
index 24447138fea2bdd7b58b1e9bac6786e016740605..8ad3c48169f3bc84c81e3f2158db55e603be8cdc 100644
--- a/tooltip/index.mjs
+++ b/tooltip/index.mjs
@@ -114,7 +114,7 @@ var Tooltip = BaseTooltip.extend('tooltip', {
           return _this.onMouseEnter(event, options);
         };
         el.$_ptooltipMouseLeaveEvent = this.onMouseLeave.bind(this);
-        el.$_ptooltipClickEvent = this.onClick.bind(this);
+        el.$_ptooltipClickEvent = (e) => this.onClick(e);
         el.addEventListener('mouseenter', el.$_ptooltipMouseEnterEvent);
         el.addEventListener('mouseleave', el.$_ptooltipMouseLeaveEvent);
         el.addEventListener('click', el.$_ptooltipClickEvent);
@@ -182,12 +182,12 @@ var Tooltip = BaseTooltip.extend('tooltip', {
     onBlur: function onBlur(event) {
       var el = event.currentTarget;
       var hideDelay = el.$_ptooltipHideDelay;
-      this.hide(el, hideDelay);
+      this.getTooltipElement(el) && this.hide(el, hideDelay);
     },
     onClick: function onClick(event) {
       var el = event.currentTarget;
       var hideDelay = el.$_ptooltipHideDelay;
-      this.hide(el, hideDelay);
+      this.getTooltipElement(el) && this.hide(el, hideDelay);
     },
     onKeydown: function onKeydown(event) {
       var el = event.currentTarget;
@@ -376,7 +376,7 @@ var Tooltip = BaseTooltip.extend('tooltip', {
       var tooltipElement = this.getTooltipElement(el);
       var arrowElement = this.getArrowElement(el);
       var hostOffset = this.getHostOffset(el);
-      var left = hostOffset.left - getOuterWidth(tooltipElement);
+      var left = Math.max(hostOffset.left + (getOuterWidth(el) - getOuterWidth(tooltipElement)) / 2, 0);
       var top = hostOffset.top + (getOuterHeight(el) - getOuterHeight(tooltipElement)) / 2;
       tooltipElement.style.left = left + 'px';
       tooltipElement.style.top = top + 'px';
